modelkit/core/models/distant_model.py

```diff
@@ -50,17 +57,25 @@ class AsyncDistantHTTPModel(AsyncModel[ItemType, ReturnType]):
     def __init__(self, **kwargs):
         super().__init__(**kwargs)
         self.endpoint = self.model_settings["endpoint"]
+        self.endpoint_params = self.model_settings.get("endpoint_params", {})
+        self.built_endpoint = build_url(self.endpoint, self.endpoint_params)
         self.aiohttp_session: Optional[aiohttp.ClientSession] = None
 
     def _load(self):
         pass
 
     @retry(**SERVICE_MODEL_RETRY_POLICY)
-    async def _predict(self, item):
+    async def _predict(self, item, **kwargs):
         if self.aiohttp_session is None:
             self.aiohttp_session = aiohttp.ClientSession()
+        endpoint_params = kwargs.get("endpoint_params")
+        built_endpoint = (
+            self.built_endpoint
+            if not endpoint_params
+            else build_url(self.endpoint, endpoint_params)
+        )
         async with self.aiohttp_session.post(
-            self.endpoint,
+            built_endpoint,
```

Comment:
We can save us the custom build_url logic if we use `params` argument here, can't we ?